packages/sync/src/class-lock.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Lock class.
+ *
+ * @package automattic/jetpack-sync
+ */
+
+namespace Automattic\Jetpack\Sync;
+
+/**
+ * Lock class
+ */
+class Lock {
+	/**
+	 * Prefix of the blog lock transient.
+	 *
+	 * @access public
+	 *
+	 * @var string
+	 */
+	const LOCK_PREFIX = 'jp_sync_lock_';
+
+	/**
+	 * Default Lifetime of the lock.
+	 *
+	 * @access public
+	 *
+	 * @var int
+	 */
+	const LOCK_TRANSIENT_EXPIRY = 15; // Seconds.
+
+	/**
+	 * Attempt to lock.
+	 *
+	 * @access public
+	 *
+	 * @param string $name lock name.
+	 * @param int    $expiry lock lifetime.
+	 *
+	 * @return boolean True if succeeded, false otherwise.
+	 */
+	public static function attempt( $name, $expiry = self::LOCK_TRANSIENT_EXPIRY ) {
```

Comment:
Instead of having static method could we change this to be more object. 
This will make it easier to test other parts of this in isolation. 

Since we can just mock the lock.

Reply:
:done: https://github.com/Automattic/jetpack/pull/14039/commits/bc9bcd69492ddbb2c1d3d9928fbcf4cdfb8f0545